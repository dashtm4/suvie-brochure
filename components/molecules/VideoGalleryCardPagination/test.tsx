import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { VideoGalleryCardPagination } from '.'

describe('<VideoGalleryCardPagination />', () => {
  const length = 2
  const activeIndex = 1
  const changeVideo = jest.fn()

  it('should renders correct <VideoGalleryCardPagination />', () => {
    const wrapper = shallow(
      <VideoGalleryCardPagination
        length={length}
        activeIndex={activeIndex}
        changeVideo={changeVideo}
      />
    )

    expect(wrapper.find('VideoGalleryCardPaginationItem')).toHaveLength(2)
    expect(
      wrapper.find('VideoGalleryCardPaginationItem').at(0).prop('isActive')
    ).toEqual(false)
    expect(
      wrapper.find('VideoGalleryCardPaginationItem').at(1).prop('isActive')
    ).toEqual(true)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <VideoGalleryCardPagination />', () => {
    const className = 'test-class-name'

    const wrapper = shallow(
      <VideoGalleryCardPagination
        length={length}
        activeIndex={activeIndex}
        changeVideo={changeVideo}
        className={className}
      />
    )

    expect(wrapper.prop('className')).toMatch(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
