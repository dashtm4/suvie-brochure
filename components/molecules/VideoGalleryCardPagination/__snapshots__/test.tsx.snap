// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoGalleryCardPagination /> should renders correct <VideoGalleryCardPagination /> 1`] = `
<div
  className="flex justify-center items-center gap-x-2"
>
  <VideoGalleryCardPaginationItem
    changeVideo={[MockFunction]}
    index={0}
    isActive={false}
    key="0"
  />
  <VideoGalleryCardPaginationItem
    changeVideo={[MockFunction]}
    index={1}
    isActive={true}
    key="1"
  />
</div>
`;

exports[`<VideoGalleryCardPagination /> should renders correct <VideoGalleryCardPagination /> 2`] = `
<div
  className="flex justify-center items-center gap-x-2 test-class-name"
>
  <VideoGalleryCardPaginationItem
    changeVideo={[MockFunction]}
    index={0}
    isActive={false}
    key="0"
  />
  <VideoGalleryCardPaginationItem
    changeVideo={[MockFunction]}
    index={1}
    isActive={true}
    key="1"
  />
</div>
`;
